src/gui/widgets/dynarec_disassembly.cc

```diff
@@ -0,0 +1,206 @@
+/***************************************************************************
+ *   Copyright (C) 2021 PCSX-Redux authors                                 *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.           *
+ ***************************************************************************/
+
+#include "gui/widgets/dynarec_disassembly.h"
+
+#include <capstone/capstone.h>
+
+#include <cinttypes>
+#include <fstream>
+
+#include "gui/gui.h"
+#include "imgui.h"
+
+int PCSX::Widgets::Disassembly::writeFile() {
```

Comment:
Two remarks here:

(1) This function returns either 0 or -1, which is essentially a boolean
(2) The return value is never checked against

(1) means we probably want to change to either a boolean or an enum (success / failure), and (2) potentially means we actually don't really care about returning anything,so, void.

Reply:
Good point. I initially had the Disassemble function return an enum and would show an error popup depending on what enum result returned. I took this out because I thought it may have been a bit over-engineered. I slimmed this down to messages in the Logs and only a popup if it was a critical error. 

I will either fix this to return a bool and check on it, or make it void and have it spit a message to logging.